custom_components/oasis_mini/pyoasiscontrol/device.py

```
@@ -0,0 +1,762 @@
+            return track.get("name", f"Unknown Title (#{self.track_id})")
+        return None
+
+    @property
+    def drawing_progress(self) -> float | None:
+        """
+        Compute drawing progress percentage for the current track.
+        
+        If the current track or its SVG content is unavailable, returns None.
+        
+        Returns:
+            progress_percent (float | None): Percentage of the drawing completed (0–100), clamped to 100; `None` if no track or SVG content is available.
```

Ruff (0.14.5)

Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)